app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java

```diff
@@ -593,6 +601,54 @@ public void notifyDataSetChanged() {
         }
     }
 
+    /**
+     * Restarts the upload process for a contribution
+     * @param contribution
+     */
+    private void restartUpload(Contribution contribution) {
+        contribution.setState(Contribution.STATE_QUEUED);
+        contributionsPresenter.saveContribution(contribution);
+        Timber.d("Restarting for %s", contribution.toString());
+    }
+
+    /**
+     * Retry for a failed contribution only if it does not exist on the server
+     * @param contribution
+     * @param retries
+     */
+    private void retryFailedUpload(Contribution contribution, int retries) {
+        String filePath = contribution.getLocalUri().getPath();
+        String fileSHA;
+       try {
+           fileSHA = fileUtilsWrapper.getSHA1(fileUtilsWrapper.getFileInputStream(filePath));
+       } catch (FileNotFoundException e) {
+           fileSHA = null;
+       }
+        boolean fileExists = mediaClient
+            .checkFileExistsUsingSha(fileSHA)
+            .subscribeOn(Schedulers.io()).blockingGet();
+        if (fileExists && !contribution.getWikidataUpdateWasSuccessful()) {
+            Timber.d("%s exists on the server already but lacks caption",
+                contribution.getMedia().getFilename());
+            // Show a popup to the user to edit the caption
+            DialogUtil.showAlertDialog(
+                getActivity(),
+                getString(R.string.wikidata_edit_failed_title),
+                getString(R.string.wikidata_edit_failed_explanation, contribution.getMedia().getFilename()),
+                getString(R.string.ok),
+                () -> {},
+                true
+            );
```

Comment:
Showing this as a dialog right away is likely not going to be good UX wise since there might be multiple uploads for which this could happen. That means the dialog would be trigger sequentially and spam the user.

Instead of that it might be good to send this as a notification (ideally in a separate channel). We could show the corresponding upload when tapping on the notification for the sake of convenience. Also, a much better enhancement would be to show this as a (dismissable) warning note in the contribution. This would be a good non-intrusive way of conveying the situation the upload to the user.

Reply:
Reverting this commit, thank you for reminding me. I'll create a separate issue for this since this is breaking some other things too.

Reply:
Could you elaborate a bit on what exactly is breaking ? 🤔

Reply:
"That means the dialog would be trigger sequentially and spam the user" <- In my experience this always happens for a single file (presumably the file that was being uploaded when Internet was cut). So I don't think we need to worry about spamming here.